typing/jkind.ml

```diff
@@ -2387,7 +2468,8 @@ let intersection_or_error ~type_equal ~jkind_of_type ~reason t1 t2 =
         history =
           combine_histories ~type_equal ~jkind_of_type reason (Pack_jkind t1)
             (Pack_jkind t2);
-        has_warned = t1.has_warned || t2.has_warned
+        has_warned = t1.has_warned || t2.has_warned;
+        quality = Not_best
```

Comment:
This is a right-kind, so this `Not_best` carries no information, right? Then I would add a

```
(* as required by the fact that this is a [jkind_r] *)
```

to this line to tell the reader to look away.